tests/5.Transaction.js

```diff
@@ -1515,5 +1520,58 @@ describe('Sign', () => {
 
       assert.deepStrictEqual(txgroup[0].group, txgroup[1].group);
     });
+    it('should be able to translate box references to encoded references', () => {
+      const testCases = [
+        [
+          [{ appIndex: 100, name: [0, 1, 2, 3] }],
+          [100],
+          9999,
+          [{ i: 1, n: [0, 1, 2, 3] }],
+        ],
+        [[], [], 9999, []],
+        [
+          [
+            { appIndex: 0, name: [0, 1, 2, 3] },
+            { appIndex: 9999, name: [4, 5, 6, 7] },
+          ],
+          [100],
+          9999,
+          [
+            { i: 0, n: [0, 1, 2, 3] },
+            { i: 0, n: [4, 5, 6, 7] },
+          ],
+        ],
+        [
+          [{ appIndex: 100, name: [0, 1, 2, 3] }],
+          [100],
+          100,
+          [{ i: 1, n: [0, 1, 2, 3] }],
+        ],
+        [
+          [
+            { appIndex: 7777, name: [0, 1, 2, 3] },
+            { appIndex: 8888, name: [4, 5, 6, 7] },
+          ],
+          [100, 7777, 8888, 9999],
+          9999,
+          [
+            { i: 2, n: [0, 1, 2, 3] },
+            { i: 3, n: [4, 5, 6, 7] },
+          ],
+        ],
+      ];
+      for (const testCase of testCases) {
+        const expected = testCase[3];
+        const actual = translateBoxReferences(
+          testCase[0],
+          testCase[1],
+          testCase[2]
+        );
+        assert.deepStrictEqual(expected.length, actual.length);
```

Comment:
Wouldn't `assert.deepStrictEqual(expected, actual)` be sufficient here instead of separating length and element-wise comparison?